django/db/backends/postgresql/operations.py

```diff
@@ -303,6 +340,13 @@ def bulk_insert_sql(self, fields, placeholder_rows):
         values_sql = ", ".join("(%s)" % sql for sql in placeholder_rows_sql)
         return "VALUES " + values_sql
 
+    def adapt_integerfield_value(self, value, internal_type):
```

Comment:
What about defining this function only for `psycopg3` at all? There is no reason to call this for `psyocpg2` just to delegated to the parent.

Reply:
But then we need another `hasattr()` check in `IntegerField.get_db_prep_value()`.

Reply:
Not sure I understand, the base database backend has an `adapt_integerfield_value` method which simply returns the value, this is callable from every backend… https://github.com/django/django/pull/15687/files/b54cb7d14096b0aaa7382480f1b56807129da623#diff-3c8a46d3879e8abd5d0b4d601b58c7acbdd6054602f5a810a7e86ae97e0d931fR528